lib/improver/standardise.py

```diff
@@ -0,0 +1,446 @@
+                cutout_coord.coord_system != grid_coord.coord_system):
+            return False
+
+        # search for cutout coordinate points in larger grid
+        cutout_start = cutout_coord.points[0]
+        find_start = [np.isclose(cutout_start, grid_point)
+                      for grid_point in grid_coord.points]
+        if not np.any(find_start):
+            return False
+
+        start = find_start.index(True)
+        end = start + len(cutout_coord.points)
+        try:
+            if not np.allclose(cutout_coord.points,
+                               grid_coord.points[start:end]):
+                return False
+        except ValueError:
+            # raised by np.allclose if "end" index overshoots edge of grid
+            # domain - slicing does not raise IndexError
+            return False
+
+    return True
+
+
+class StandardiseGridAndMetadata(BasePlugin):
+
+    """Plugin to regrid cube data and standardise metadata"""
+
+    REGRID_REQUIRES_LANDMASK = {"bilinear": False,
+                                "nearest": False,
+                                "nearest-with-mask": True}
+
+    def __init__(self, regrid_mode='bilinear', extrapolation_mode='nanmask',
+                 landmask=None, landmask_vicinity=25000):
+        """
+        Initialise regridding parameters
+
+        Args:
+            regrid_mode (str):
+                Mode of interpolation in regridding.
+            extrapolation_mode (str):
+                Mode to fill regions outside the domain in regridding.
+            landmask (iris.cube.Cube or None):
+                Land-sea mask ("land_binary_mask") on the input cube grid.
+                Required for "nearest-with-mask" regridding option.
+            landmask_vicinity (float):
+                Radius of vicinity to search for a coastline, in metres
+
+        Raises:
+            ValueError: If a landmask is required but not passed in
+        """
+        if landmask is None and "nearest-with-mask" in regrid_mode:
+            msg = ("An argument has been specified that requires an input "
+                   "landmask cube but none has been provided")
+            raise ValueError(msg)
+        self.regrid_mode = regrid_mode
+        self.extrapolation_mode = extrapolation_mode
+        self.landmask_source_grid = landmask
+        self.landmask_vicinity = (
+            None if landmask is None else landmask_vicinity)
+        self.landmask_name = "land_binary_mask"
+
+        # set attributes to inherit from the target grid
+        self.grid_attributes = [
+            'mosg__grid_version', 'mosg__grid_domain', 'mosg__grid_type']
+
+    def _adjust_landsea(self, cube, target_grid):
+        """
+        Adjust regridded data using differences between the target landmask
+        and that obtained by regridding the source grid landmask, to ensure
+        that the "land" or "sea" nature of the points in the regridded cube
+        matches that of the target grid.
+
```

Comment:
It wasn't intentional, particularly.  I don't think the static methods need an args section (they're all quite small and self-explanatory), but I can add one to the slightly longer functions if you think it would be helpful.

Reply:
Yes, `Args` sections would be useful.

Reply:
OK.  Added.